services/src/main/java/io/druid/cli/CliBroker.java

```diff
@@ -120,10 +130,52 @@ public void configure(Binder binder)
             MetricsModule.register(binder, CacheMonitor.class);
 
             LifecycleModule.register(binder, Server.class);
+
+            binder.bind(ForSideEffectsOnlyProvider.Child.class).toProvider(ForSideEffectsOnlyProvider.class).asEagerSingleton();
           }
         },
         new LookupModule(),
         new SqlModule()
     );
   }
+
+  private static class ForSideEffectsOnlyProvider implements Provider<ForSideEffectsOnlyProvider.Child>
+  {
+    final static class Child {};
+
+    @Inject
+    public ForSideEffectsOnlyProvider(DruidNodeAnnouncer announcer, @Self DruidNode druidNode,
+                                      LookupNodeService lookupNodeService, Lifecycle lifecycle)
+    {
+      DiscoveryDruidNode discoveryDruidNode = new DiscoveryDruidNode(druidNode,
```

Comment:
Do this in the `get()` call instead of here.  Setting this stuff up in the constructor (especially when marked as an eagerSingleton) can lead to weird and interesting interleaving of injections.